### ISSUE TITLE
Allow running programms/commands when XIVLauncher starts

### ISSUE THREAD

**Issue by s-renz** [goatcorp/FFXIVQuickLauncher#1452]
### Update disclaimer

- [X] Yes, I have checked and my request is not related to the game updating and plugins not working correctly.

### What happened, what did not, what did you want to happen?

XIVLauncher already has an option to start other programms or commands when the game launches. 

Could this maybe be extended to running stuff when XIVLauncher has been started to run along side the launcher?

This could be used to start for example gaming VPNs which need to run before the game, ACT (which also needs to be started before the game unless you use Deucalion injection) or even triggering auto OTP-entering (Bitwarden for Windows has a CLI client which can be used to send the OTP directly to XIVLauncher via a script).

### Version

Standard version (Windows, Lutris)

### Platform

Applies to all

**Comment by kaicataldo** [comment]
I’d be willing to take a crack at implementing this feature if this is something the maintainers want and we can decide on the interface for it. My naive take is that it would be nice to have a notion of “hooks,” (“on start,” “on quit”, “on password retrieval”, etc.), but I’m also new around here so am happy to defer to the consensus of the team.
 
 
Ultimately, I’d love to have a way for the launcher to be able to retrieve my password from something safer than a plaintext file, whether it’s through some blessed 3rd party option, a built-in feature, or having a feature like hooks that allow an arbitrary script to be run at given times in the launcher’s lifecycle.

Thanks!

**Comment by reiichi001** [comment]
> ACT (which also needs to be started before the game unless you use Deucalion injection) 

This is incorrect. ACT does not need to be running before the game, even when using packet captures instead of Deucalion (which a user should be user by default anyways). It needs to be running before you connect to the Lobby/World servers, which means it can be started on the Title screen.
